src/main/java/hu/oe/nik/szfmv/environment/models/Road.java

```diff
@@ -1,6 +1,92 @@
 package hu.oe.nik.szfmv.environment.models;
 
+import java.awt.*;
+import java.awt.geom.AffineTransform;
+import java.util.HashMap;
+import java.util.Map;
+
 public class Road extends Crossable {
+    private static HashMap<String, Polygon> roadPolyMap;
+
+    static {
+        roadPolyMap = new HashMap<String, Polygon>();
+        roadPolyMap.put(
+                "road_2lane_90left.png",
+                new Polygon(
+                        //525x525
+                        new int[]{0, 143, 227, 307, 358, 408, 461, 502, 525, 175, 162, 140, 97, 48, 0},
```

Comment:
Ezen tömbök minden elemére a checkstyle magic number hibát dob... teljesen egyértelmű, hogy ezekre szükség van, de meg lehet oldani úgy, hogy az előbbi probléma se álljon fenn. A Checkstyle nem ugrik rá az adattagok deklarálásakor definiált konstansokra. Az alábbi esetben nem nyivákol:

```java
//525x525
private static final Polygon ROAD_2LANE_90_LEFT = new Polygon(
        new int[]{0, 143, 227, 307, 358, 408, 461, 502, 525, 175, 162, 140, 97, 48, 0},
        new int[]{0, 21, 51, 96, 138, 197, 272, 364, 525, 525, 461, 422, 380, 356, 350},
        15);
```
*(Azt, hogy privát legyen-e, te jobban átlátod, én hirtelen privát konstansként venném föl őket.)*

Rá fogok nézni, hogy ez a static blokkos megoldás mennyire zavar be itt a kódstílus kérdésben, ugyanakkor ha ezt félretesszük is átláthatóbbnak gondolom, hogy a poligonok „fönt” konstansként vannak definiálva és a HashMap deklarációjánál csak hivatkozva vannak. Ezáltal kisebb, kompaktabb az a rész, nem pedig ilyen ömlesztett.

Reply:
Rendben, mindenképp megnézzük. Köszi!